plsync/mlabconfig.py

```diff
@@ -45,13 +64,152 @@ def parse_flags():
+def export_server_records_v6(output, sites, decoration=''):
+  comment(output, 'hosts v6%s' % (' decorated' if decoration else ''))
+  for site in sites:
+      # TODO: change site['nodes'] to a pre-sorted list type.
+      for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+          if node.ipv6_is_enabled():
+              write_aaaa_record(
+                  output, node.recordname(decoration), node.ipv6())
+
+
+def export_experiment_records(output, sites, experiments):
+  for experiment in experiments:
+    if experiment['index'] is None:
+      # Ignore experiments without an IP address.
+      continue
+    export_experiment_records_v4(output, sites, experiment)
+    export_experiment_records_v4(output, sites, experiment, decoration='v4')
+
+    export_experiment_records_v6(output, sites, experiment)
+    export_experiment_records_v6(output, sites, experiment, decoration='v6')
+
+
+def export_experiment_records_v4(output, sites, experiment, decoration=''):
+  comment(output, '%s v4%s' % (experiment.dnsname(), (
+      ' decorated' if decoration else '')))
+  for site in sites:
+    # TODO: change site['nodes'] to a pre-sorted list type.
+    for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+        # TODO: should sitenames exclude mlab4's?
+        write_a_record(output, experiment.sitename(node, decoration),
+                       experiment.ipv4(node))
+        write_a_record(output, experiment.recordname(node, decoration),
+                       experiment.ipv4(node))
+
+
+def export_experiment_records_v6(output, sites, experiment, decoration=''):
+  comment(output, '%s v6%s' % (experiment.dnsname(), (
+      ' decorated' if decoration else '')))
+  for site in sites:
+    # TODO: change site['nodes'] to a pre-sorted list type.
+    for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+      # TODO: should sitenames exclude mlab4's?
+      if (node.ipv6_is_enabled() and experiment.ipv6(node)):
+        write_aaaa_record(output, experiment.sitename(node, decoration),
+                          experiment.ipv6(node))
+        write_aaaa_record(output, experiment.recordname(node, decoration),
+                          experiment.ipv6(node))
+
+
+def export_mlab_zone_records(output, sites, experiments):
+    export_router_and_switch_records(output, sites)
+    export_pcu_records(output, sites)
+    export_server_records(output, sites)
+    export_experiment_records(output, sites, experiments)
+
+
+def serial_rfc1912(ts):
+    """Returns an rfc1912 style serial id (YYYYMMDDnn) for a DNS zone file."""
+    # RFC1912 (http://www.ietf.org/rfc/rfc1912.txt) recommends 'nn' as the
+    # revision. However, identifying and incrementing this value is a manual,
+    # error prone step. Instead, the following calculates 'nn' from HH:MM (00:00
+    # to 23:59, or 0 to 1439) by calculating the corresponding value in the
+    # range 00 to 99. 'nn' increases by 1 about every 15 minutes.
```

Comment:
Rather than adopting a convention, would it not be more fail-safe and obviate the need for remembering or documenting anything, by just having the code pick an appropriate ID?  It is almost unthinkable that 100 changes could happen in the course of 24 hours.  However, it is not unthinkable at all that two changes could happen in the course of 15 minutes, even though unusual.  Most likely this would happen in a case where we publish some DNS change, and then immediately realize we forgot to add something else, or see an error so publish again just a few minutes later.   There is nothing that says we have to stick to the RFC's conventions for serial numbers, right?  How about just using the current Unix timestamp for the serial number?  Not only is it guaranteed to be unique every second, but it can readily be converted to a precise date, which gives us even more granular data on when a given change was published.